src/components/Checkboxes.tsx

```diff
@@ -0,0 +1,296 @@
+  onChange?: (values: string[]) => void
+): JSX.Element => {
+  let id;
+  if (item.id) {
+    id = item.id;
+  } else {
+    id = `${idPrefix}-${item.value.replace(/\s+/g, "-").toLowerCase()}`;
+  }
+  const itemHintId = `${id}-item-hint`;
+  let currentValues = items
+    .filter(item => item.checked)
+    .map(item => item.value);
+  return (
+    <React.Fragment key={index}>
+      <div className="govuk-checkboxes__item">
+        <input
+          className="govuk-checkboxes__input"
+          id={id}
+          name={name}
+          type="checkbox"
+          value={item.value}
+          checked={item.checked}
+          disabled={item.disabled}
+          data-aria-controls={
+            item.childrenWhenChecked ? `conditional-${id}` : undefined
+          }
+          aria-describedby={item.hint?.children ? itemHintId : undefined}
+          {...checkboxExtraAttributes}
+          onChange={
+            onChange !== undefined
+              ? (event: ChangeEvent<HTMLInputElement>): void => {
+                  let newValues = currentValues;
+                  const target = event.target;
+                  if (target && target.checked) {
+                    if (!currentValues.includes(target.value)) {
+                      newValues.push(target.value);
+                    }
+                  } else {
+                    newValues = newValues.filter(v => v !== target.value);
+                  }
+                  currentValues = newValues;
+                  onChange(newValues);
+                }
+              : undefined
+          }
+        />
+        <Label
+          className={classNames(
+            "govuk-checkboxes__label",
+            item.label.className
+          )}
+          labelFor={id}
+          {...item.label}
+        >
+          {item.label.children}
+        </Label>
+        {item.hint?.children && (
+          <Hint
+            id={itemHintId}
+            className={classNames(
+              "govuk-checkboxes__hint",
+              item.hint?.className
+            )}
+          >
+            {item.hint?.children}
+          </Hint>
+        )}
+      </div>
+      {item.childrenWhenChecked && (
+        <div
+          className={classNames("govuk-checkboxes__conditional", {
+            "govuk-checkboxes__conditional--hidden": !item.checked
+          })}
+          id={`conditional-${id}`}
+        >
+          {item.childrenWhenChecked}
+        </div>
+      )}
+    </React.Fragment>
+  );
+};
+
+const renderItem = (
+  items: Checkbox[],
+  item: Checkbox,
+  name: string,
+  index: number,
+  idPrefix: string,
+  onChange?: (values: string[]) => void
+): JSX.Element => {
+  const checkboxExtraAttributes = Attributes.ariaAndData(item);
+  return renderCheckbox(
+    items,
+    item,
+    name,
+    index,
+    idPrefix,
+    checkboxExtraAttributes,
+    onChange
+  );
+};
+
+export const Checkboxes = (props: CheckboxesProps): JSX.Element => {
```

Comment:
```suggestion
export const Checkboxes: React.FunctionComponent<CheckboxesProps> = props => {
```

I think the function will infer the types if you do this, and you'll get typechecking on your `propTypes`.